tests/integration-tests/tests/ad_integration/test_ad_integration.py

```diff
@@ -323,7 +323,8 @@ def _directory_factory(
     yield _directory_factory
 
     for region, stack_dict in created_directory_stacks.items():
-        for stack_type, stack_name in stack_dict.items():
+        for stack_type in ["nlb", "directory"]:
```

Comment:
Add a brief comment on why `nlb` needs to be deleted first so that it's clear why it's written this way over the former.